librz/search/search.c

```diff
@@ -790,31 +790,25 @@ RZ_IPI int rz_search_hit_cmp(RZ_NULLABLE RzSearchHit *a, RZ_NULLABLE RzSearchHit
 /**
  * \brief      Allocate and initialize a new RzSearchHit
  *
- * \param[in]  hit_desc  The hit description linked to the hit (can be NULL)
- * \param[in]  address   The address where the hit happened
- * \param[in]  size      The size of the hit data (can be 0)
+ * \param[in]  hit_desc    The hit description linked to the hit (can be NULL)
+ * \param[in]  address     The address where the hit happened
+ * \param[in]  size        The size of the hit data (can be 0)
+ * \param[in]  hit_detail  Contains additional information about the hit (can be NULL)
  *
  * \return     On success returns a valid pointer, otherwise NULL
  */
-RZ_IPI RZ_OWN RzSearchHit *rz_search_hit_new(const char *hit_desc, ut64 address, size_t size, const char *hit_comment) {
+RZ_IPI RZ_OWN RzSearchHit *rz_search_hit_new(const char *hit_desc, ut64 address, size_t size, RzSearchHitDetail *hit_detail) {
```

Comment:
```suggestion
RZ_IPI RZ_OWN RzSearchHit *rz_search_hit_new(const char *hit_desc, ut64 address, size_t size, RZ_OWN RzSearchHitDetail *hit_detail) {
```

Reply:
Rzown should not go there, but nullable should be there.

Reply:
Both then? Because it assigns the pointer directly and not cloning the content. And frees it later. Or was I looking incorrectly?

Reply:
ok